src/Mod/PartDesign/App/FeatureExtrude.cpp

```diff
@@ -598,179 +665,124 @@ App::DocumentObjectExecReturn* FeatureExtrude::buildExtrusion(ExtrudeOptions opt
-                }
-                // open the shell by removing the furthest face
-                upToShape = makeShellFromUpToShape(upToShape, sketchshape, dir);
-            }
-
-            if (!supportface.hasSubShape(TopAbs_WIRE)) {
-                supportface = TopoShape();
-            }
-            if (legacyPocket) {
-                auto mode =
-                    base.isNull() ? TopoShape::PrismMode::None : TopoShape::PrismMode::CutFromBase;
-                prism = base.makeElementPrismUntil(sketchshape,
-                                                   supportface,
-                                                   upToShape,
-                                                   dir,
-                                                   mode,
-                                                   false /*CheckUpToFaceLimits.getValue()*/);
-                // DO NOT assign id to the generated prism, because this prism is
-                // actually the final result. We obtain the subtracted shape by cut
-                // this prism with the original base. Assigning a minus self id here
-                // will mess up with preselection highlight. It is enough to re-tag
-                // the profile shape above.
-                //
-                // prism.Tag = -this->getID();
-
-                // And the really expensive way to get the SubShape...
-                try {
-                    TopoShape result(0, getDocument()->getStringHasher());
-                    if (base.isNull()) {
-                        result = prism;
-                    }
-                    else {
-                        result.makeElementCut({base, prism});
-                    }
-
-                    // store shape before refinement
-                    this->rawShape = result;
-                    result = refineShapeIfActive(result);
-                    this->AddSubShape.setValue(result);
-                }
-                catch (Standard_Failure&) {
-                    return new App::DocumentObjectExecReturn(
-                        QT_TRANSLATE_NOOP("Exception", "Up to face: Could not get SubShape!"));
-                }
+        }
 
-                if (getAddSubType() == Additive) {
-                    prism = base.makeElementFuse(this->AddSubShape.getShape());
+        // --- Combine generated prisms (all in global CS) ---
+        TopoShape prism(0, getDocument()->getStringHasher());
+        if (prisms.empty()) {
+            return new App::DocumentObjectExecReturn(QT_TRANSLATE_NOOP("Exception", "No extrusion geometry was generated."));
+        }
+        else if (prisms.size() == 1) {
+            prism = prisms[0];
+        }
+        else {
+            try {
+                // For "2 sides" resulting in two shapes, perform XOR: (A U B) - (A n B)
+                // We don't have a OpCodes for XOR so we need to do it manually?
+                TopoShape prism_A = prisms[0];
+                TopoShape prism_B = prisms[1];
```

Comment:
Can you elaborate on why XOR is needed here?

Reply:
For the case where both sides are on the same side aof the sketch. So 2 sides, both length, one is 10mm the other 5mm. Expected result is from 5 to 10mm But if we used OR, then we would get from 0 to 10mm.

Reply:
Let me clarify the situation above because it wasn't so clear. Say you set 2 lengths : 
Length 1 = 10 mm
Length 2 = -5 mm

What the user expect when doing this is that the extrusion starts at 10mm and stops at 5mm, ie the sketch is outside of the extrusion.

Reply:
You are correct it would be better to introduce opcode XOR.

Reply:
I will do this then

Reply:
<img width="988" height="197" alt="image" src="https://github.com/user-attachments/assets/b703ae13-1dcb-4b57-a0b4-6a28d1225e07" />

note OCC does not have a XOR API. So I will implement in toposhape basically the same as I have here. ie use a fuse common and cut

Reply:
> Let me clarify the situation above because it wasn't so clear. Say you set 2 lengths : Length 1 = 10 mm Length 2 = -5 mm
> 
> What the user expect when doing this is that the extrusion starts at 10mm and stops at 5mm, ie the sketch is outside of the extrusion.

Now indeed it is a lot clearer, thanks!